### ISSUE TITLE
Changes takes forever to commit

### ISSUE THREAD

**Issue by bharathy84** [RepreZen/KaiZen-OpenAPI-Editor#427]
Whenever I make any changes (Even add a simple space), and save the document, Spring Tool Suite (STS) opens a window saying "User Operation is Waiting" and list lot of changes in progress. Some are listed below.

> The user operation is waiting for "Update KaiZen Editor validation markers" to complete.
> 
> JPA Project Change Event Handler (Waiting)
> Update Kaizen Editor Validation markers
> Waiting User Operation.

A simple one line change takes lot of time to save... A big change take an hour to save

**Comment by bharathy84** [comment]
No one facing this issue?

**Comment by tedepstein** [comment]
@bharathy84 , we haven't tested specifically with STS, so it's possible that something is specific to that environment.  We don't normally expect validations to take long. 

Are you working with a very large source YAML file, or set of files? Do your source files have external reference properties, and if so have you verified that all of those resources are accessible...?

**Comment by bharathy84** [comment]
Yes its a large file with references to other files. Is there a way to disable validation markers. Yaml changes are saved quickly in Yedit editor.

**Comment by tedepstein** [comment]
AFAIK we don't currently have a way to disable validation. Are the references to other files:
* within the Eclipse workspace; or
* on the local filesystem, but outside the workspace; or
* on a remote server, accessed through HTTP?

**Comment by bharathy84** [comment]
The references are within same workspace. I am facing this issue in swagger 2.0.

For 3.0, I am facing a new issue from yesterday. Larger yaml files are taking too long to open.

STS becomes unresponsive for long time.

**Comment by bharathy84** [comment]
Swagger 2.0 in STS is taking too long to validate changes.
Swagger 3.0 in STS is fine

**Comment by manzoo-dev** [comment]
Noticed this issue with STS (Oxygen base) and also Eclipse Oxygen release.
The only way to combat this that i can see is closing the project and workspace and reopening it after every 30 mins or so. A big hindrance on the development flow of things.
The yaml files are about 3000 lines, which is not exceptionally large. All references are with in the same workspace and the workspace has only 2 projects.

**Comment by ghost** [comment]
Similar problem here. Although some of our referenced files are remote. Validation hangs, sometimes indefinitely.
